### PR TITLE
improve refund page with payment provider specific information (bug 1123470)

### PR DESCRIPTION
**Pull request by andymckay**
Lessons learned from the first refund, we need a bit more information on the refund page. This adds in information that Bango would like to know in terminology that Bango understands.

I'll do another PR with the package details to add to this.

screenshot 2015-01-19 16 28 04

### CODE REVIEW

File: media/css/devreg/manage.styl [hunk 1]
```diff
@@ -364,3 +364,7 @@ td .checkbox-choices:not(.regions) {
 .island-error {
     margin-bottom: 10px;
 }
+
+#transaction-lookup {
+    dd: 5px 0px 5px;
```
Comment (robhudson) [inline media/css/devreg/manage.styl]: I think something is missing here. `dd` isn't a CSS property.

File: mkt/constants/payments.py [hunk 2]
```diff
@@ -97,6 +97,7 @@
 )
 
 PROVIDER_LOOKUP = dict([(v, k) for k, v in PROVIDER_CHOICES])
+PROVIDER_LOOKUP_INVERTED = dict([(k, v) for k, v in PROVIDER_CHOICES])
```
Comment (robhudson) [inline mkt/constants/payments.py]: The one above it is inverted. This one looks like no change?

File: mkt/lookup/templates/lookup/transaction_summary.html [hunk 3]
```diff
@@ -37,12 +38,51 @@
...
+          <p>{{ _('Currently unable to retrieve transaction status.') }}</p>
+        {% endif %}
+
+        {% if provider == 'bango' %}
+          <p>
+            {% trans bango_support='support@bango.com' %}
+            This payment was made through Bango. To get more information
+            contact <a href="mailto:{{ bango_support }}">Bango support</a> with the
+            following information.
+            {% endtrans %}
```
Comment (robhudson) [inline mkt/lookup/templates/lookup/transaction_summary.html]: Add an indent level inside the `{% trans %}`?

File: mkt/lookup/templates/lookup/transaction_summary.html [hunk 4]
```diff
@@ -37,12 +38,51 @@
...
+          </p>
+        {% endif %}
+
+        {% if provider == 'reference' %}
+          <p>
+            {% trans %}
+            This payment was made through the <b>reference implementation</b>.
+            It wasn't a real transaction.
+            {% endtrans %}
```
Comment (robhudson) [inline mkt/lookup/templates/lookup/transaction_summary.html]: Same here? Indent?

File: mkt/lookup/templates/lookup/transaction_summary.html [hunk 5]
```diff
@@ -37,12 +38,51 @@
...
+        {% endif %}
+
+        {% if lookup['transaction'] %}
+          <dl>
+            <dt>{{ _('Amount paid') }}</dt>
+            <dd>{{ amount }} {{ currency }}</dd>
+            <dt>{{ _('Lookup id') }}</dt>
+            <dd>{{ support }}</dd>
+            <dt>{{ _('Timestamp') }}</dt>
+            <dd>{{ timestamp }} PST</dd>
```
Comment (robhudson) [inline mkt/lookup/templates/lookup/transaction_summary.html]: We record it in PST? Not UCT?

File: mkt/lookup/views.py [hunk 6]
```diff
@@ -128,25 +130,44 @@ def _transaction_summary(tx_uuid):
     refund_contribs = contrib.get_refund_contribs()
     refund_contrib = refund_contribs[0] if refund_contribs.exists() else None
 
+
+    lookup = {'status': True, 'transaction': True}
+    pay = {}
+    try:
+        pay = client.api.generic.transaction.get_object_or_404(
+                uuid=contrib.transaction_id)
```
Comment (robhudson) [inline mkt/lookup/views.py]: 4-space indent here?

Comment (andymckay) [inline mkt/lookup/views.py]: will do

File: mkt/lookup/templates/lookup/transaction_summary.html (hunk 5 above)
Comment (andymckay) [inline mkt/lookup/templates/lookup/transaction_summary.html]: currently yes, see https://bugzilla.mozilla.org/show_bug.cgi?id=863436

File: mkt/lookup/templates/lookup/transaction_summary.html (hunk 3 above)
Comment (andymckay) [inline mkt/lookup/templates/lookup/transaction_summary.html]: I should

File: media/css/devreg/manage.styl (hunk 1 above)
Comment (andymckay) [inline media/css/devreg/manage.styl]: Oh right, thanks, hmm.

File: mkt/constants/payments.py (hunk 2 above)
Comment (andymckay) [inline mkt/constants/payments.py]: I guess you are right, I looked at the one above and wanted the opposite. Its actually just `dict(PROVIDER_CHOICES)` as well.

**Comment by robhudson** [comment]
r+

**Pull request merged**